<a href="https://colab.research.google.com/github/Florani1/Hamoye-Internship-Codes/blob/main/PremierProject.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [ ]:
df = pd.read_excel('/content/Subnational African Data_Aug.19.2013.xls')

In [ ]:
print(df.shape)

(471, 68)


There are 471 observations and 68 variables.

In [ ]:
df.head()

Country name ISO code  ISO numeric code Subnational region name  \
0      Burundi      BDI               108                 Bubanza   
1      Burundi      BDI               108        Bujumbura Mairie   
2      Burundi      BDI               108         Bujumbura Rural   
3      Burundi      BDI               108                  Bururi   
4      Burundi      BDI               108                 Cankuzo   

   CCAPS subnational region ID  Year Data Source             Website  \
0                       108002  2011         DHS  www.measuredhs.com   
1                       108003  2011         DHS  www.measuredhs.com   
2                       108004  2011         DHS  www.measuredhs.com   
3                       108005  2011         DHS  www.measuredhs.com   
4                       108006  2011         DHS  www.measuredhs.com   

   Net primary attendance rate (%)  Primary NAR standard error  ...  \
0                        62.212669                    1.541257  ...   
1                        75.802948                    1.635965  ...   
2                        62.212669                    1.541257  ...   
3                        66.308830                    1.372277  ...   
4                        61.176640                    1.455065  ...   

   Electricity (% of hh) sample size  Radio in household (% of households)  \
0                             1441.0                             50.203342   
1                              912.0                             75.303909   
2                             1441.0                             50.203342   
3                             1921.0                             51.190273   
4                             2405.0                             51.336479   

   Radio (% of hh) standard error  Radio (% of hh) sample size  \
0                        1.984068                       1441.0   
1                        2.757758                        912.0   
2                        1.984068                       1441.0   
3                        1.153134                       1921.0   
4                        1.450344                       2405.0   

   Television in household (% of households)  \
0                                   1.656027   
1                                  44.327900   
2                                   1.656027   
3                                   3.321293   
4                                   1.289977   

   Television (% of hh) standard error  Television (% of hh) sample size  \
0                             0.644799                            1441.0   
1                             5.504890                             912.0   
2                             0.644799                            1441.0   
3                             1.444279                            1921.0   
4                             0.369800                            2405.0   

   Radio and/or television in household (% of households)  \
0                                          49.767658        
1                                          77.148560        
2                                          49.767658        
3                                          51.439995        
4                                          51.364285        

   Radio and/or TV (% of hh) standard error  \
0                                  1.906297   
1                                  2.845784   
2                                  1.906297   
3                                  1.158005   
4                                  1.450414   

   Radio and/or TV (% of hh) sample size  
0                                 1441.0  
1                                  912.0  
2                                 1441.0  
3                                 1921.0  
4                                 2405.0  

[5 rows x 68 columns]

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 471 entries, 0 to 470
Data columns (total 68 columns):
 #   Column                                                  Non-Null Count  Dtype  
---  ------                                                  --------------  -----  
 0   Country name                                            471 non-null    object 
 1   ISO code                                                471 non-null    object 
 2   ISO numeric code                                        471 non-null    int64  
 3   Subnational region name                                 471 non-null    object 
 4   CCAPS subnational region ID                             471 non-null    int64  
 5   Year                                                    471 non-null    int64  
 6   Data Source                                             471 non-null    object 
 7   Website                                                 471 non-null    object 
 8   Net primary attendance rate (%)         

In [ ]:
df.isnull().sum()

Country name                                               0
ISO code                                                   0
ISO numeric code                                           0
Subnational region name                                    0
CCAPS subnational region ID                                0
                                                          ..
Television (% of hh) standard error                       10
Television (% of hh) sample size                          10
Radio and/or television in household (% of households)    10
Radio and/or TV (% of hh) standard error                  10
Radio and/or TV (% of hh) sample size                     10
Length: 68, dtype: int64

In [ ]:
df.nunique()

Country name                                               38
ISO code                                                   38
ISO numeric code                                           38
Subnational region name                                   445
CCAPS subnational region ID                               471
                                                         ... 
Television (% of hh) standard error                       395
Television (% of hh) sample size                          327
Radio and/or television in household (% of households)    398
Radio and/or TV (% of hh) standard error                  398
Radio and/or TV (% of hh) sample size                     324
Length: 68, dtype: int64

The primary key column in this dataset is CCAPS Subnational region ID



A total of 38 countries is involved in this dataset

In [ ]:
df.isnull().sum()

Country name                                               0
ISO code                                                   0
ISO numeric code                                           0
Subnational region name                                    0
CCAPS subnational region ID                                0
                                                          ..
Television (% of hh) standard error                       10
Television (% of hh) sample size                          10
Radio and/or television in household (% of households)    10
Radio and/or TV (% of hh) standard error                  10
Radio and/or TV (% of hh) sample size                     10
Length: 68, dtype: int64

In [ ]:
(df.isnull().sum()/(len(df)))*100

Country name                                              0.000000
ISO code                                                  0.000000
ISO numeric code                                          0.000000
Subnational region name                                   0.000000
CCAPS subnational region ID                               0.000000
                                                            ...   
Television (% of hh) standard error                       2.123142
Television (% of hh) sample size                          2.123142
Radio and/or television in household (% of households)    2.123142
Radio and/or TV (% of hh) standard error                  2.123142
Radio and/or TV (% of hh) sample size                     2.123142
Length: 68, dtype: float64

**To remove rows with null values**

In [ ]:
df.dropna(inplace=True)

In [ ]:
df.isnull().sum()

Country name                                              0
ISO code                                                  0
ISO numeric code                                          0
Subnational region name                                   0
CCAPS subnational region ID                               0
                                                         ..
Television (% of hh) standard error                       0
Television (% of hh) sample size                          0
Radio and/or television in household (% of households)    0
Radio and/or TV (% of hh) standard error                  0
Radio and/or TV (% of hh) sample size                     0
Length: 68, dtype: int64

**All null values have now been removed**

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 326 entries, 0 to 366
Data columns (total 68 columns):
 #   Column                                                  Non-Null Count  Dtype  
---  ------                                                  --------------  -----  
 0   Country name                                            326 non-null    object 
 1   ISO code                                                326 non-null    object 
 2   ISO numeric code                                        326 non-null    int64  
 3   Subnational region name                                 326 non-null    object 
 4   CCAPS subnational region ID                             326 non-null    int64  
 5   Year                                                    326 non-null    int64  
 6   Data Source                                             326 non-null    object 
 7   Website                                                 326 non-null    object 
 8   Net primary attendance rate (%)         

**To drop unneeded columns**

In [ ]:
df.drop(['Data Source','Website'], axis=1, inplace=True)

In [ ]:
df.drop(['Country name'], axis=1, inplace=True)

In [ ]:
df.head()

ISO code  ISO numeric code Subnational region name  \
0      BDI               108                 Bubanza   
1      BDI               108        Bujumbura Mairie   
2      BDI               108         Bujumbura Rural   
3      BDI               108                  Bururi   
4      BDI               108                 Cankuzo   

   CCAPS subnational region ID  Year  Net primary attendance rate (%)  \
0                       108002  2011                        62.212669   
1                       108003  2011                        75.802948   
2                       108004  2011                        62.212669   
3                       108005  2011                        66.308830   
4                       108006  2011                        61.176640   

   Primary NAR standard error  Primary NAR sample size  \
0                    1.541257                   1494.0   
1                    1.635965                    718.0   
2                    1.541257                   1494.0   
3                    1.372277                   2036.0   
4                    1.455065                   2213.0   

   Gross primary attendance rate (%)  Primary GAR standard error  ...  \
0                         102.472580                    3.636928  ...   
1                         112.636269                    4.013371  ...   
2                         102.472580                    3.636928  ...   
3                         117.472328                    2.735463  ...   
4                         113.127411                    3.140667  ...   

   Electricity (% of hh) sample size  Radio in household (% of households)  \
0                             1441.0                             50.203342   
1                              912.0                             75.303909   
2                             1441.0                             50.203342   
3                             1921.0                             51.190273   
4                             2405.0                             51.336479   

   Radio (% of hh) standard error  Radio (% of hh) sample size  \
0                        1.984068                       1441.0   
1                        2.757758                        912.0   
2                        1.984068                       1441.0   
3                        1.153134                       1921.0   
4                        1.450344                       2405.0   

   Television in household (% of households)  \
0                                   1.656027   
1                                  44.327900   
2                                   1.656027   
3                                   3.321293   
4                                   1.289977   

   Television (% of hh) standard error  Television (% of hh) sample size  \
0                             0.644799                            1441.0   
1                             5.504890                             912.0   
2                             0.644799                            1441.0   
3                             1.444279                            1921.0   
4                             0.369800                            2405.0   

   Radio and/or television in household (% of households)  \
0                                          49.767658        
1                                          77.148560        
2                                          49.767658        
3                                          51.439995        
4                                          51.364285        

   Radio and/or TV (% of hh) standard error  \
0                                  1.906297   
1                                  2.845784   
2                                  1.906297   
3                                  1.158005   
4                                  1.450414   

   Radio and/or TV (% of hh) sample size  
0                                 1441.0  
1                                  912.0  
2                                 1441.0  
3   

In [ ]:
df.describe()

ISO numeric code  CCAPS subnational region ID         Year  \
count        326.000000                   326.000000   326.000000   
mean         489.187117                489196.306748  2008.475460   
std          238.856633                238856.809033     2.089758   
min          108.000000                108002.000000  2003.000000   
25%          288.000000                288004.250000  2007.000000   
50%          487.000000                487005.000000  2008.000000   
75%          686.000000                686008.750000  2010.000000   
max          894.000000                894009.000000  2011.000000   

       Net primary attendance rate (%)  Primary NAR standard error  \
count                       326.000000                  326.000000   
mean                         67.709036                    2.802090   
std                          19.450906                    1.417553   
min                          19.211811                    0.502372   
25%                          54.770161                    1.616250   
50%                          73.775112                    2.505683   
75%                          83.660372                    3.541417   
max                          95.413872                    7.193609   

       Primary NAR sample size  Gross primary attendance rate (%)  \
count               326.000000                         326.000000   
mean               1087.165644                          99.154915   
std                1060.771927                          25.445972   
min                 212.000000                          25.051022   
25%                 678.000000                          85.932434   
50%                 900.000000                         102.600609   
75%                1228.500000                         116.893064   
max               14398.000000                         150.556335   

       Primary GAR standard error  Primary GAR sample size  \
count                  326.000000               326.000000   
mean                     4.493296              1089.966258   
std                      1.866895              1061.924454   
min                      1.029757               212.000000   
25%                      3.152429               678.250000   
50%                      3.892934               900.000000   
75%                      5.535463              1243.000000   
max                     10.625566             14405.000000   

       Net secondary attendance rate (%)  ...  \
count                         326.000000  ...   
mean                           27.919773  ...   
std                            18.028179  ...   
min                             1.837479  ...   
25%                            14.305085  ...   
50%                            24.035156  ...   
75%                            38.406351  ...   
max                            78.802788  ...   

       Electricity (% of hh) sample size  \
count                         326.000000   
mean                         1109.524540   
std                           903.970723   
min                           324.000000   
25%                           770.250000   
50%                           924.500000   
75%                          1210.000000   
max                         12090.000000   

       Radio in household (% of households)  Radio (% of hh) standard error  \
count                            326.000000                      326.000000   
mean                              58.948763                        2.690094   
std                               15.437248                        1.122522   
min                               13.913477                        0.797229   
25%                               47.460161                        1.951249   
50%                               60.218603                        2.529208   
75%                               70.597902                        3.066482   
max                               89.975082                        8.593001   

       Radio (% of hh)

### **STATISTICAL ANALYSIS BEGINS**

**From the stastical summary above, we can infer the following for the Primary level:**


**Year ranges from 2003 - 2011.**


**NPAR ranges from 19.2% - 95.4%.**



**NPAR error ranges from 0.5%-7.2%. This shows that the estimate is reliable.**


**NAR Sample size ranges from 212 to 14398.**
**Average students who participated in the survey are 1087.**
**The medium number of students who participated are 900.**


**GPAR ranges from 25.05% - 150.56%**

**GPAR error ranges from 1.0%-10.6%**.
**The average is 4.49%, so the estimate is still reliable.**
**The median is 3.89%, so we have more reliable estimate.**

**GAR Sample size ranges from 212 to 14405.**

**Created below a subset dataframe for secondary school.**

In [ ]:
df_subset = df[['Net secondary attendance rate (%)','Secondary NAR standard error','Secondar NAR sample size','Gross secondary attendance rate (%)','Secondary GAR standard error','Secondary GAR sample size']]
secondary = df_subset.describe()

In [ ]:
secondary

Net secondary attendance rate (%)  Secondary NAR standard error  \
count                         326.000000                    326.000000   
mean                           27.919773                      3.073192   
std                            18.028179                      1.415611   
min                             1.837479                      0.603776   
25%                            14.305085                      2.102229   
50%                            24.035156                      2.926195   
75%                            38.406351                      4.036954   
max                            78.802788                      8.510447   

       Secondar NAR sample size  Gross secondary attendance rate (%)  \
count                326.000000                           326.000000   
mean                 700.116564                            45.295048   
std                  462.014724                            28.260953   
min                  136.000000                             6.007842   
25%                  423.250000                            24.141747   
50%                  624.500000                            38.124609   
75%                  784.750000                            58.317075   
max                 4842.000000                           145.247025   

       Secondary GAR standard error  Secondary GAR sample size  
count                    326.000000                 326.000000  
mean                       4.733186                 703.128834  
std                        2.265847                 463.704205  
min                        1.335817                 136.000000  
25%                        3.201640                 424.250000  
50%                        4.224002                 627.000000  
75%                        5.763951                 793.000000  
max                       14.922399                4853.000000

**From the stastical summary above, we can infer the following for the Secondary** **level:**

**NSAR ranges from 1.8% - 78.8%.**
**The average is 27.9% while the median is 24%. This shows that there are fewer level of attendance rate in secondary schools compared to primary schools.**

**NSAR error ranges from 0.6%-8.5%. The average is 3.07 while the median is 2.93. This shows that the estimate is reliable.**

**SNAR Sample size ranges from 136 to 4842. Average students who participated in** **the survey are 700.**
**These shows that we have fewer participants in secondary schools compared to primary schools.**

**GSAR ranges from 6% - 145.2%**

**GSAR error ranges from 1.3%-14.9%. The average is 4.7%.**

**SGAR Sample size ranges from 136 to 4853.**
**These also shows that we have fewer participants in secondary schools compared to primary schools.**

**Created below a subset dataframe for Literacy Rate.**

In [ ]:
df1_subset = df[['Literacy rate (15 & over)','Literacy rate (15 & over) standard error','Literacy rate (15 & over) size','Literacy rate (15-24)','Literacy rate (15-24) standard error','Literacy rate (15-24) size','Literacy rate (15-19)','Literacy rate (15-19) standard error','Literacy rate (15-19) size','Literacy rate (25-49)','Literacy rate (25-49) standard error','Literacy rate (25-49) size']]

In [ ]:
Literacy=df1_subset.describe()

In [ ]:
Literacy

Literacy rate (15 & over)  Literacy rate (15 & over) standard error  \
count                 347.000000                                347.000000   
mean                   61.434973                                  2.627702   
std                    25.397349                                  1.517131   
min                     9.497565                                  0.153905   
25%                    43.382933                                  1.586654   
50%                    66.046158                                  2.336796   
75%                    81.646332                                  3.522886   
max                    98.898605                                  8.128676   

       Literacy rate (15 & over) size  Literacy rate (15-24)  \
count                      347.000000             347.000000   
mean                      1806.498559              67.072803   
std                       1507.064101              24.195768   
min                        350.000000              10.415296   
25%                       1105.500000              51.189569   
50%                       1393.000000              71.427422   
75%                       1970.000000              87.865166   
max                      13994.000000              99.634010   

       Literacy rate (15-24) standard error  Literacy rate (15-24) size  \
count                            347.000000                  347.000000   
mean                               2.989891                  852.760807   
std                                1.717665                  732.352751   
min                                0.107816                  104.000000   
25%                                1.775956                  434.500000   
50%                                2.742748                  650.000000   
75%                                4.024814                 1030.500000   
max                                9.101959                 6015.000000   

       Literacy rate (15-19)  Literacy rate (15-19) standard error  \
count             347.000000                            347.000000   
mean               71.134001                              3.523134   
std                22.232864                              2.063453   
min                12.650104                              0.193470   
25%                57.099506                              2.024794   
50%                75.423141                              3.241239   
75%                90.226864                              4.448523   
max                99.664429                             12.317142   

       Literacy rate (15-19) size  Literacy rate (25-49)  \
count                  347.000000             347.000000   
mean                   374.775216              57.286808   
std                    301.652187              27.039922   
min                     52.000000               4.976739   
25%                    217.500000              36.393225   
50%                    293.000000              61.075508   
75%                    425.500000              77.263718   
max                   3119.000000              98.958878   

       Literacy rate (25-49) standard error  Literacy rate (25-49) size  
count                            347.000000                  347.000000  
mean                               2.787651                 1023.190202  
std                                1.542747                  790.544181  
min                                0.176317                  226.000000  
25%                                1.697982                  634.500000  
50%                                2.410761                  826.000000  
75%                                3.693015                 1160.000000  
max                                9.235173                 8181.000000

**The summary above shows that we have more literates between the ages of 15-19. The record shows an average of 71.1% while the average literacy rate for ages 25-49 is 57.3%**

In [ ]:
df2_subset = df[['Access to improved water (% of population)','Improved Water (% of pop) standard error','Improved Water (% of pop) sample size','Access to improved sanitation (% of population)','Improved Sanitation (% of pop) standard error','Improved Sanitation (% of pop) sample size','Electricity in household (% of population)','Electricity (% of pop) standard error','Electricity (% of pop) sample size']]

In [ ]:
Wse=df2_subset.describe()

In [ ]:
Wse

Access to improved water (% of population)  \
count                                  470.000000   
mean                                    60.442817   
std                                     24.733905   
min                                      0.000000   
25%                                     41.069965   
50%                                     64.240891   
75%                                     79.161705   
max                                    100.000000   

       Improved Water (% of pop) standard error  \
count                                470.000000   
mean                                   4.163505   
std                                    2.741863   
min                                    0.000000   
25%                                    1.854867   
50%                                    3.808712   
75%                                    5.802433   
max                                   14.950254   

       Improved Water (% of pop) sample size  \
count                             470.000000   
mean                             5306.778723   
std                              3981.937583   
min                               427.000000   
25%                              3432.750000   
50%                              4582.000000   
75%                              6119.750000   
max                             54841.000000   

       Access to improved sanitation (% of population)  \
count                                       470.000000   
mean                                         40.165780   
std                                          29.219446   
min                                           0.000000   
25%                                          15.250207   
50%                                          35.081390   
75%                                          59.746507   
max                                         100.000000   

       Improved Sanitation (% of pop) standard error  \
count                                     470.000000   
mean                                        2.955834   
std                                         1.995369   
min                                         0.000000   
25%                                         1.547232   
50%                                         2.621436   
75%                                         4.171311   
max                                         9.014301   

       Improved Sanitation (% of pop) sample size  \
count                                  470.000000   
mean                                  5302.602128   
std                                   3981.623067   
min                                    427.000000   
25%                                   3432.750000   
50%                                   4553.500000   
75%                                   6113.250000   
max                                  54831.000000   

       Electricity in household (% of population)  \
count                                  449.000000   
mean                                    28.916043   
std                                     30.222012   
min                                      0.000000   
25%                                      5.400793   
50%                                     14.685029   
75%                                     49.463890   
max                                    100.000000   

       Electricity (% of pop) standard error  \
count                             449.000000   
mean                                2.885390   
std                                 2.477114   
min                                 0.000000   
25%                                 0.949273   
50%                                 1.995803   
75%                                 4.241601   
max                                13.054160   

       Electricity (% of pop) sample size  
count                          449.000000  
mean                          5296.064588  
std                           4027.472117  
min                            427.000000  

**From the summary above, average of 60% has access to improved water,average of **40% has access to improved sanitation and average of 28.9% has Electricity.
**This shows that we majority of the population does not have electricity.**

In [ ]:
df3_subset = df[['Radio in household (% of population)','Radio (% of hh) standard error','Radio (% of hh) sample size','Television in household (% of households)','Television (% of hh) standard error','Television (% of hh) sample size','Radio and/or television in household (% of households)','Radio and/or TV (% of hh) standard error','Radio and/or TV (% of hh) sample size']]


In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 471 entries, 0 to 470
Data columns (total 68 columns):
 #   Column                                                  Non-Null Count  Dtype  
---  ------                                                  --------------  -----  
 0   Country name                                            471 non-null    object 
 1   ISO code                                                471 non-null    object 
 2   ISO numeric code                                        471 non-null    int64  
 3   Subnational region name                                 471 non-null    object 
 4   CCAPS subnational region ID                             471 non-null    int64  
 5   Year                                                    471 non-null    int64  
 6   Data Source                                             471 non-null    object 
 7   Website                                                 471 non-null    object 
 8   Net primary attendance rate (%)         

**Created below a subset dataset for the Literacy Rate.**

In [ ]:
df1_subset = df[[''